### PR TITLE
Support for AWS IAM RolesAnywhere - cross-cloud access

### PR DESCRIPTION
**Pull request by orishoshan**
This PR adds support for creating AWS IAM RolesAnywhere profiles based on a configured RolesAnywhere Trust Anchor (CA cert) and SPIFFE trust domain, both managed by cert-manager.

When combined with the credentials-operator and the instructions in https://github.com/otterize/otterize-csi-spiffe-demo, this can be used to achieve simple cross-cloud access.

This also temporarily brings back the old code that was moved to credentials.go in the intents-operator.

Related:
* https://github.com/otterize/intents-operator/pull/372
* https://github.com/otterize/helm-charts/pull/183

### CODE REVIEW

File: src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	logger.WithField("arn", *role.Arn).Info("created AWS role for ServiceAccount")
+
+	// update annotation if it doesn't exist or if it is misconfigured
+	shouldUpdate := !ok || roleARN != *role.Arn
+
+	return shouldUpdate, role, nil
+}
+
+func (r *ServiceAccountReconciler) shouldUseSoftDeleteStrategy(serviceAccount *corev1.ServiceAccount) bool {
+	if r.markRolesAsUnusedInsteadOfDelete {
```
Comment (omris94) [inline src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go:170]: I suggest changing the order of the "if global UseSoftDelete" and the "if per pod useSoft" statements here. This way, when both configurations are present, the "per pod" configuration will be given priority over the global one. It makes more sense. 
(I know it's my code 😅)

File: src/operator/controllers/aws_iam/sa_pod_webhook/sa_pod_webhook.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+package sa_pod_webhook
```
Comment (orishoshan) [inline src/operator/controllers/aws_iam/sa_pod_webhook/sa_pod_webhook.go:1]: This is the original pod webhook, from before the refactor.

File: src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
+package serviceaccount_old
```
Comment (orishoshan) [inline src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go:1]: The old reconciler.

File: src/operator/controllers/aws_iam/spiffe_rolesanywhere_pod_webhook/spiffe_pod_webhook.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
+package spiffe_rolesanywhere_pod_webhook
```
Comment (orishoshan) [inline src/operator/controllers/aws_iam/spiffe_rolesanywhere_pod_webhook/spiffe_pod_webhook.go:1]: The new webhook that creates SPIFFE identities with the csi-driver and works with RolesAnywhere.

File: src/operator/controllers/iam/serviceaccounts/serviceaccount_controller_test.go [hunk 5]
```diff
@@ -252,36 +251,37 @@ func (s *TestServiceAccountSuite) TestServiceAccountSuite_CreateIAMRole_NoArn_Pa
...
-			return nil
-		},
-	)
-
-	s.mockIAM.EXPECT().OnServiceAccountUpdate(gomock.Any(), gomock.AssignableToTypeOf(&serviceAccount)).Return(false, false, nil)
-
-	_, err := s.reconciler.Reconcile(context.Background(), req)
-	s.Require().NoError(err)
-}
+//
```
Comment (orishoshan) [inline src/operator/controllers/iam/serviceaccounts/serviceaccount_controller_test.go:254]: Normally I would delete this, but left this here to make it easier to revive it in a few days

File: src/operator/operatorconfig/config.go [hunk 6]
```diff
@@ -1,20 +1,92 @@
...
-	EnvPrefix = "OTTERIZE"
+const (
+	CertProviderSPIRE       = "spire"
+	CertProviderCloud       = "otterize-cloud"
+	CertProviderCertManager = "cert-manager"
+	CertProviderNone        = "none"
 )
 
 func init() {
+	viper.SetDefault(EnableAWSServiceAccountManagementKey, EnableAWSServiceAccountManagementDefault)
```
Comment (orishoshan) [inline src/operator/operatorconfig/config.go:54]: Moved CLI flags to environment variables, but kept CLI flags for backwards compatibility. Haven't changed helm chart.

**Review by omris94 — APPROVED** [review]

File: src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+	isNoLongerReferencedByPodsOrIsBeingDeleted := serviceAccount.DeletionTimestamp != nil || !isReferencedByPods
+
+	if isNoLongerReferencedByPodsOrIsBeingDeleted {
+		err = r.awsAgent.DeleteOtterizeIAMRole(ctx, req.Namespace, req.Name)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to remove service account: %w", err)
+		}
+
+		if viper.GetBool(operatorconfig.EnableAWSRolesAnywhereKey) {
```
Comment (omris94) [inline src/operator/controllers/aws_iam/serviceaccount_old/serviceaccount_controller.go:79]: Maybe it is worth aligning it with `markRolesAsUnusedInsteadOfDelete` (it doesn't matter if doing it with struct field or with viper, just use the same method for both flags)

**Pull request merged**